src/entry.rs

```diff
@@ -6,7 +6,25 @@ const VALUE_POINTER: u8 = 1 << 1;
 pub struct Entry {
     pub key: Bytes,
     pub value: Bytes,
-    pub meta: u8,
+    pub(crate) meta: u8,
+    pub user_meta: u8,
+    pub expires_at: u64,
+    pub(crate) version: u64,
+}
+
+pub struct EntryRef<'a> {
```

Comment:
Why not just use `&Entry`?

Reply:
`Entry` uses `Bytes` internally. But sometimes we could only get `&[u8]` (if not copy).

Reply:
FYI, https://github.com/tikv/agatedb/blob/develop/src/value.rs#L185

The key and value is a `Vec<u8>` (or you may think of it as `BytesMut`), which will only be valid for one iteration. This way, we let our user to decide whether to copy it into a `Bytes`.

Reply:
Got it.